src/OrchardCore/OrchardCore.Data.YesSql/OrchardCoreBuilderExtensions.cs

```diff
@@ -143,37 +144,41 @@ public static OrchardCoreBuilder AddDataAccess(this OrchardCoreBuilder builder)
                 services.AddScoped(sp =>
                 {
                     var store = sp.GetService<IStore>();
-
                     if (store == null)
                     {
                         return null;
                     }
 
                     var session = store.CreateSession();
-
                     var scopedServices = sp.GetServices<IScopedIndexProvider>();
-
                     session.RegisterIndexes(scopedServices.ToArray());
 
                     ShellScope.Current
                         .RegisterBeforeDispose(scope =>
                         {
-                            return scope.ServiceProvider
-                                .GetRequiredService<IDocumentStore>()
-                                .CommitAsync();
+                            return session.SaveChangesAsync();
                         })
                         .AddExceptionHandler((scope, e) =>
                         {
-                            return scope.ServiceProvider
-                                .GetRequiredService<IDocumentStore>()
-                                .CancelAsync();
+                            return session.CancelAsync();
                         });
 
                     return session;
                 });
 
-                services.AddScoped<IDocumentStore, DocumentStore>();
-                services.AddSingleton<IFileDocumentStore, FileDocumentStore>();
+                services.AddScoped<IDocumentStore>(sp =>
+                {
+                    var session = sp.GetService<ISession>();
+                    if (session == null)
+                    {
+                        return null;
+                    }
+
+                    return new DatabaseDocumentStore(session);
```

Comment:
Why not register `DatabaseDocumentStore` as scoped, and it would resolve the session itself.

Reply:
For some reason, it's required by autosetup.

Reply:
We need to understand why it was not an issue before this PR which was registering `DocumentStore` (now named `DatabaseDocumentStore`) like I suggested.